src/main/scala/de/upb/cs/swt/delphi/crawler/Configuration.scala

```diff
@@ -20,13 +20,36 @@ import java.net.URI
 
 import akka.stream.ThrottleMode
 import com.sksamuel.elastic4s.ElasticsearchClientUri
+import de.upb.cs.swt.delphi.crawler.instancemanagement.InstanceRegistry
+import de.upb.cs.swt.delphi.crawler.io.swagger.client.model.Instance
+import de.upb.cs.swt.delphi.crawler.io.swagger.client.model.InstanceEnums.ComponentType
 
 import scala.concurrent.duration._
+import scala.util.{Failure, Success}
+
+class Configuration {
+
+  lazy val elasticsearchClientUri: ElasticsearchClientUri = ElasticsearchClientUri({
+    if(elasticsearchInstance.portnumber.isEmpty){
+      elasticsearchInstance.iP.getOrElse("elasticsearch://localhost:" + defaultElasticSearchPort)
+    }else{
+      elasticsearchInstance.iP.getOrElse("elasticsearch://localhost") + ":" + elasticsearchInstance.portnumber.getOrElse(defaultElasticSearchPort)
+    }
+  })
+
+  lazy val elasticsearchInstance : Instance = InstanceRegistry.retrieveElasticSearchInstance(this) match {
+    case Success(instance) => instance
+    case Failure(_) => Instance(
+      None,
+      Some(sys.env.getOrElse("DELPHI_ELASTIC_URI","elasticsearch://localhost")),
```

Comment:
In the original code there was a port given. Does it work without giving it the default port?